pithos/StationsDialog.py

```diff
@@ -153,15 +155,41 @@ def refresh_stations(self, widget):
 
     @GtkTemplate.Callback
     def add_station_cb(self, dialog, response):
-        logging.info("in add_station_cb {} {}".format(dialog.result, response))
+        self.result = dialog.result
+        logging.info("in add_station_cb {} {}".format(self.result, response))
         if response == Gtk.ResponseType.OK:
-            self.worker_run("add_station_by_music_id", (dialog.result.musicId,),
+            self.worker_run("add_station_by_music_id", (self.result.musicId,),
                             self.station_added, "Creating station...")
         dialog.hide()
         dialog.destroy()
         self.searchDialog = None
 
+    def station_already_exists(self, station):
+        # May as well reuse the delete_confirm_dialog. All we need is some text and a Yes, No.
```

Comment:
Code is read by humans too :P

It can probably done in pure Python, it is little more than a default Gtk.MessageDialog.